OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -64,6 +64,12 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 			if (!attack.Info.IgnoresVisibility && type == TargetType.Actor && !Target.Actor.HasTrait<FrozenUnderFog>() && !self.Owner.CanTargetActor(Target.Actor))
 				return NextActivity;
 
+			// Drop the target if it is being captured by our own team.
+			var capturable = Target.Actor.TraitOrDefault<Capturable>();
+			var capturesinfo = Target.Actor.TraitOrDefault<CapturesInfo>();
```

Comment:
Yep, this is always true or false. You will have to check against `CaptureThreshold` and hit points. I am not sure if the complication is really required here. When someone wants to capture, just assume everyone should stop firing.